File: backend/Restaurant-Project/src/main/java/com/morgan/restaurant/model/Order.java
package com.morgan.restaurant.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import net.minidev.json.annotate.JsonIgnore;

import javax.persistence.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "orders")
public class Order extends CategoryOrder{

    @Column(name = "price")
    private int price;

    @Column(name = "image")
    private String img;


    @Lob
    @Column(name = "description")
    private String description;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @ManyToOne
    @JoinColumn(name = "id_category")
    private Category category;
}

File: frontend/src/app/model/order.ts
export class Order {

    id: number;
    name: string;
    data_created: Date;
    data_updated: Date;
    description: string;
    img: string;
    price: number;
    //make constructor
    constructor(id: number, name: string, data_created: Date, data_updated: Date, description: string, image: string, price: number) {
        this.id = id;
        this.name = name;
        this.data_created = data_created;
        this.data_updated = data_updated;
        this.description = description;
        this.img = image;
        this.price = price;
    }
    
}
